galsim/utilities.py

```diff
@@ -1821,3 +1821,68 @@ def __exit__(self, type, value, traceback):
         if self.reverse:  # pragma: no cover
             ps = ps.reverse_order()
         ps.print_stats(self.nlines)
+
+
+def least_squares(fun, x0, args=(), kwargs={}, max_iter=1000, tol=1e-9, lambda_init=1.0):
+    """Perform a non-linear least squares fit using the Levenberg-Marquardt algorithm.
+
+    Drop in replacement for scipy.optimize.least_squares when using default options,
+    though many fewer options available in general.
+
+    Parameters:
+        fun: Function which computes vector of residuals, with the signature
+             fun(params, *args, **kwargs) -> np.ndarray.
+        x0: Initial guess for the parameters.
+        args: Additional arguments to pass to the function.
+        kwargs: Additional keyword arguments to pass to the function.
+        max_iter: Maximum number of iterations.  [default: 1000]
+        tol: Tolerance for convergence.  [default: 1e-9]
+        lambda_init: Initial damping factor for Levenberg-Marquardt.  [default: 1.0]
+
+    Returns:
+        A named tuple with fields:
+            x: The final parameter values.
+            cost: The final cost (sum of squared residuals).
+    """
+    from collections import namedtuple
+    params = np.array(x0)
+    lambda_ = lambda_init
+
+    for _ in range(max_iter):
```

Comment:
The code coverage miss is this loop never ending.  Should either add a test with a low max_iter, (e.g. 1 probably), or add `#pragma: no branch` to this line.